lms/djangoapps/learner_dashboard/tests/test_programs.py

```diff
@@ -0,0 +1,138 @@
+
+    def _get_program_checklist(self, program_id):
+        '''
+            The convenience function to get all the program related page element we would like to check against
+        '''
+        return [
+            self.PROGRAM_NAMES[program_id],
+            self._get_program_url(self.PROGRAMS_API_RESPONSE['results'][program_id]['marketing_slug']),
+            self.PROGRAMS_API_RESPONSE['results'][program_id]['organizations'][0]['display_name'],
+        ]
+
+    @httpretty.activate
+    def test_get_program_with_no_enrollment(self):
+        response = self._setup_and_get_program()
+        for program_element in self._get_program_checklist(0):
+            self.assertNotIn(program_element, response.content)
+        for program_element in self._get_program_checklist(1):
+            self.assertNotIn(program_element, response.content)
+
+    @httpretty.activate
+    def test_get_one_program(self):
+        self._create_course_and_enroll(self.student, *self.COURSE_KEYS[0].split('/'))
+        response = self._setup_and_get_program()
+        for program_element in self._get_program_checklist(0):
+            self.assertIn(program_element, response.content)
+        for program_element in self._get_program_checklist(1):
+            self.assertNotIn(program_element, response.content)
+
+    @httpretty.activate
+    def test_get_both_program(self):
+        self._create_course_and_enroll(self.student, *self.COURSE_KEYS[0].split('/'))
+        self._create_course_and_enroll(self.student, *self.COURSE_KEYS[5].split('/'))
+        response = self._setup_and_get_program()
+        for program_element in self._get_program_checklist(0):
+            self.assertIn(program_element, response.content)
+        for program_element in self._get_program_checklist(1):
+            self.assertIn(program_element, response.content)
+
+    def test_get_programs_dashboard_not_enabled(self):
+        self.create_programs_config(enable_student_dashboard=False)
+        self.client.login(username=self.student.username, password=self.PASSWORD)
+        response = self.client.get(reverse("program_listing_view"))
+        self.assertEqual(response.status_code, 404)
+
+    def test_xseries_advertise_disabled(self):
+        self.create_programs_config(xseries_ad_enabled=False)
+        self.client.login(username=self.student.username, password=self.PASSWORD)
+        response = self.client.get(reverse("program_listing_view"))
+        self.assertEqual(response.status_code, 200)
+        x_series_url = urljoin(settings.MKTG_URLS.get('ROOT'), 'xseries')
+        self.assertNotIn(x_series_url, response.content)
+
+    def test_get_programs_not_logged_in(self):
+        self.create_programs_config()
+        response = self.client.get(reverse("program_listing_view"))
+        self.assertEqual(response.status_code, 302)
+        self.assertIsInstance(response, HttpResponseRedirect)
+        self.assertIn('login', response.url)  # pylint: disable=E1101
```

Comment:
When silencing pylint, use the message name - in this case, `no-member` - instead of the message code. It's more readable.